src/System.Net.Http/tests/FunctionalTests/HttpContentTest.cs

```diff
@@ -430,6 +432,21 @@ public async Task ReadAsStringAsync_SetNoCharset_DefaultCharsetUsed()
             Assert.Equal(sourceString, result);
         }
 
+        [Fact]
```

Comment:
Also test empty quotes, and only leading, or only trailing?

Reply:
>Also test empty quotes, and only leading, or only trailing?

Agreed.  This Fact could be a Theory with more test cases.

Reply:
I spent some time writing a test for this case. I was a little surprised by how things work, so I'll summarize here.

In the current implementation, there are essentially four cases for choosing the charset:

(1) No charset is provided or there is no content-type header.

-  In this case we default to UTF-8, as per the spec.

(2) A charset is provided in the content-type header, and we recognize the charset.
- In this case, we use the provided charset.

(3) A charset is provided in the content-type header, and we do not recognize the charset.
- In this case we throw an exception indicating an invalid charset.

(4) A charset is provided that is so wrong that the whole content-type header is malformed
- In this case we throw out the whole header and decode the content as UTF-8


The cases listed here are all case (4), and actually result in the code being parsed as UTF-8

If we agree that is the correct behavior I will go ahead and pin it with a test or two. Does that sound reasonable @danmosemsft and @davidsh?

Reply:
>If we agree that is the correct behavior I will go ahead and pin it with a test or two. Does that sound reasonable @danmosemsft and @davidsh?

Yes it sounds reasonable. The low-risk path here is to write some tests to pin the current behavior as well as add functionality to handle the quoted string tokens (both properly quoted and improperly quoted).